api/v1/tigerastatus_types.go

```diff
@@ -104,6 +104,27 @@ type TigeraStatusList struct {
 	Items           []TigeraStatus `json:"items"`
 }
 
+// TigeraStatusReason represents the reason for a particular condition. A reason may be one of the below.
+type TigeraStatusReason string
+
+const (
+	AllObjectsAvailable       TigeraStatusReason = "AllObjectsAvailable"
+	ResourceNotReady          TigeraStatusReason = "ResourceNotReady"
+	PodFailure                TigeraStatusReason = "PodFailure"
+	CertificateError          TigeraStatusReason = "CertificateError"
+	InvalidConfigurationError TigeraStatusReason = "InvalidConfigurationError"
```

Comment:
I was wondering if we could consider merging InvalidConfigurationError and ResourceValidationError or if there are situations where this is not possible.

Reply:
Marking as done